src/pages/settings/Security/CloseAccountPage.js

```diff
@@ -37,105 +40,94 @@ const defaultProps = {
     },
 };
 
-class CloseAccountPage extends Component {
-    constructor(props) {
-        super(props);
+function CloseAccountPage (props) {
+    const [isConfirmModalVisible, setConfirmModalVisibility] = useState(false);
+    const [reasonForLeaving, setReasonForLeaving] = useState('');
 
-        this.onConfirm = this.onConfirm.bind(this);
-        this.validate = this.validate.bind(this);
-        this.hideConfirmModal = this.hideConfirmModal.bind(this);
-        this.showConfirmModal = this.showConfirmModal.bind(this);
-        CloseAccount.clearError();
-        this.state = {
-            isConfirmModalVisible: false,
-            reasonForLeaving: '',
+    useEffect(() => {
+        CloseAccount.setDefaultData();
```

Comment:
It's similar to what we were doing previously (though I don't love that we were doing it previously) but mostly the react docs suggest you should typically do something in the first part and undo it in the 2nd part so figure this was more inline with the "react way" but I don't have a super strong opinion, I think either is fine.

Reply:
oh after looking again... it jogged my memory a little bit.

I did it this way since that's how it was working before but ideally what we'd do is hook this up to the component directly in the `withOnyx` and pass these as actual props.

Since we were trying to change a little as possible I figured better to follow what was happening then we can clean everything else up after we're done with the refactor. 

Happy to change it still.